### PR TITLE
Turn off "var" suggestion when not apparent

### PR DESCRIPTION
**Pull request by davkean**
See: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md.

> We only use var when it's obvious what the variable type is (i.e. `var stream = new FileStream(...)` not `var stream = OpenStandardInput()`).

### CODE REVIEW

**Comment by davkean** [comment]
Note, I had to turn it off for built-in types because of https://github.com/dotnet/roslyn/issues/23714.

File: .editorconfig [hunk 1]
```diff
@@ -50,9 +50,9 @@ dotnet_style_explicit_tuple_names = true:suggestion
 # CSharp code style settings:
 [*.cs]
 # Prefer "var" everywhere
-csharp_style_var_for_built_in_types = true:suggestion
+csharp_style_var_for_built_in_types = false:none
 csharp_style_var_when_type_is_apparent = true:suggestion
-csharp_style_var_elsewhere = true:suggestion
+csharp_style_var_elsewhere = false:none
```
Comment (sharwell) [inline .editorconfig]: :memo: 'none' enforcement means that the user is free to choose to use var or not use var in the scenario. Be prepared to not comment during code reviews for the usage of var when this option is used.

**Comment by sharwell** [comment]
:bulb: Unless you are planning to write a custom set of analyzers to cover the rules for 'var' within this project, I would recommend slightly modifying the project's preferences to fit within one of the supported configurations provided by .editorconfig. The most likely scenario would be the following:

```
csharp_style_var_for_built_in_types = false:suggestion
csharp_style_var_when_type_is_apparent = true:suggestion
csharp_style_var_elsewhere = false:suggestion
```

File: .editorconfig [hunk 2]
```diff
@@ -50,9 +50,9 @@ dotnet_style_explicit_tuple_names = true:suggestion
 # CSharp code style settings:
 [*.cs]
 # Prefer "var" everywhere
```
Comment (sharwell) [inline .editorconfig]: ❕ This comment is no longer correct

File: .editorconfig [hunk 3]
```diff
@@ -50,9 +50,9 @@ dotnet_style_explicit_tuple_names = true:suggestion
...
-csharp_style_var_for_built_in_types = true:suggestion
+csharp_style_var_for_built_in_types = false:none
```
Comment (jmarolf) [inline .editorconfig]: I think we at least want suggestion here.  Probably a longer discussion about the level of enforcement. Warning could work as well

Comment (davkean) [reply .editorconfig]: Yeah I was confused how this worked - I didn't realize saying false here actually means the inverse and hence enforce it.

Comment (jmarolf) [reply .editorconfig]: I think the wording I chose here has confused a lot of people.  How would you expect this to work for enforcing this option?  What words would make sense to you?

Comment (davkean) [reply .editorconfig]: No the naming had nothing to do it with - I just have too much history and remember the old incarnation of this where when you switched off the option in Options and it didn't enforce the reverse.

Comment (davkean) [reply .editorconfig]: I tried "warning" it was much too aggressive, especially given how much code was already written like this.

**Pull request merged**